lib/controllers/facebook-login.js

```diff
@@ -75,7 +75,11 @@ module.exports = function (req, res) {
             return oauth.errorResponder(req, res, err);
           }
 
-          var nextUrl = url.parse(req.query.next || '').path || (resp.created ? config.web.register.nextUri : config.web.login.nextUri);
+          var nextUrl = oauth.common.consumeRedirectUri(req, res);
+
+          if (!nextUrl) {
+            nextUrl = url.parse(req.query.next || '').path || (resp.created ? config.web.register.nextUri : config.web.login.nextUri);
```

Comment:
That is a valid point and I think you're right about it. I'll probably just remove it.